### ISSUE TITLE
[proposal] Extend key types for BLS keys

### ISSUE THREAD

**Issue by iliecirciumaru** [uport-project/veramo#838]
I am going to integrates BLS | BBS suites. To implement things correctly, I would like to extend [key types]( https://github.com/uport-project/veramo/blob/next/packages/core/src/types/IIdentifier.ts#L51)

from 
```
export type TKeyType = 'Ed25519' | 'Secp256k1' | 'X25519'
```
to
```
export type TKeyType = 'Ed25519' | 'Secp256k1' | 'X25519' | 'Bls12381G1' | 'Bls12381G2'
```

Types of the keys are determined in the [W3C standard](https://w3c-ccg.github.io/ldp-bbs2020/#bls-12-381-g1-public-key)

**Comment by mirceanis** [comment]
Wonderful!

Do you think this integration can be done in a multi-platform way?
Ideally, the core packages should work on frontend(eg. react projects), backend(nodejs) and mobile(react-native).
We've had some issues with BLS implementations that were not suitable for this.

In any case, the `TKeyType` should indeed be extended to include a few more key types, even if they are not yet implemented or supported by our simple key management system yet. Other KMS implementations might support them.

**Comment by iliecirciumaru** [comment]
I have done some prototypes which confirmed my ideas, that integration can be done. I start today to make everything "correct" that is why decided to extend the TKeyType. BLS for TS comes from Ursa lib through JavaScript binding, so I believe it will work on majority of platforms.

**Comment by mirceanis** [comment]
this was closed in #839 but without actual support for the new key types.

@iliecirciumaru is your work on integrating BBS suites public?
